freeipa/src/main/java/com/sequenceiq/freeipa/flow/stack/stop/StackStopService.java

```diff
@@ -0,0 +1,55 @@
+package com.sequenceiq.freeipa.flow.stack.stop;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopInstancesResult;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.DetailedStackStatus;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.flow.stack.StackFailureEvent;
+import com.sequenceiq.freeipa.flow.stack.StackStartStopService;
+import com.sequenceiq.freeipa.service.stack.StackUpdater;
+
+@Component
+public class StackStopService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackStopService.class);
+
+    @Inject
+    private StackUpdater stackUpdater;
+
+    @Inject
+    private StackStartStopService stackStartStopService;
+
+    public void startStackStop(StackStopContext context) {
+        Stack stack = context.getStack();
+        if (isStopPossible(stack)) {
```

Comment:
This is confusing to me. This controls whether the status can be changed to in progress, however I do not see anything that stops the flow from going forward if this condition did not meet.